File: backend/app/schemas/book.py
from pydantic import BaseModel
from typing import Optional

class BookBase(BaseModel):
    title: str
    author: str
    genre: str
    rating: float
    year: Optional[int] = None
    image_url: Optional[str] = None
    description: Optional[str] = None


class BookCreate(BookBase):
    pass

class BookUpdate(BaseModel):
    title: Optional[str] = None
    author: Optional[str] = None
    genre: Optional[str] = None
    rating: Optional[float] = None
    year: Optional[int] = None
    image_url: Optional[str] = None
    description: Optional[str] = None

class Book(BookBase):
    id: int

    # USER SPECIFIC FIELDS
    reading_status: Optional[str] = None
    user_rating: Optional[float] = None

    class Config:
        from_attributes = True

File: frontend/src/types/book.ts
export interface Book {
  id: number;
  title: string;
  author: string;
  genre: string;
  rating: number;
  year?: number;
  image_url?: string;
  description?: string;

  // user specific fields
  reading_status?: "None" | "want_to_read" | "currently_reading" | "finished";
  user_rating?: number | null;
}

export interface BookFilters {
  genre: string;
  minRating: number;
  yearRange: string;
  search: string;
}
